### PR TITLE
Enable force_pull on Token API to validate Cluster Token on startup (OSIO#2304)

### PR DESCRIPTION
**Pull request by xcoulon**
Add the `force_pull=true` query parameter when retrieving the user's token
from `auth`.
Update the tests to use `go-vcr`.

Also: remove unused `WhoAmI` function (for good this time?)

Fixes openshiftio/openshift.io#2304

Signed-off-by: Xavier Coulon <xcoulon@redhat.com>

### CODE REVIEW

File: token/service.go [hunk 1]
```diff
@@ -38,7 +38,8 @@ func (s *tokenService) ResolveUserToken(ctx context.Context, target, token strin
 					Value: token,
 					Type:  "Bearer"}}})
 
-	res, err := client.RetrieveToken(ctx, authclient.RetrieveTokenPath(), target, nil)
+	forcePull := true
```
Comment (aslakknutsen) [inline token/service.go]: We don't want to forcePull user tokens, only cluster tokens.

File: test/data/openshift/get_user.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+---
```
Comment (aslakknutsen) [inline test/data/openshift/get_user.yaml]: What is this used for?

File: token/service.go (hunk 1 above)
Comment (xcoulon) [reply token/service.go]: ah, damn'it, I misunderstood :/ let me change that

File: test/data/openshift/get_user.yaml (hunk 2 above)
Comment (xcoulon) [reply test/data/openshift/get_user.yaml]: what part of testing the `whoami` function (older version of the parent issue), but I forgot to remove it. Let me do it now.

File: token/service.go (hunk 1 above)
Comment (alexeykazakov) [reply token/service.go]: Oh, I misunderstood the requirement for this too. If you need to check the cluster (dsaas user) token and not the user tokens then force_pull is currently ignored for cluster tokens. But I've just created a pull request to add force_pull support for cluster tokens in the Auth service - https://github.com/fabric8-services/fabric8-auth/pull/351

**Comment by fabric8cd** [comment]
@xcoulon snapshot fabric8-tenant image is available for testing.  `docker pull docker.io/fabric8/fabric8-tenant:SNAPSHOT-PR-537-4`

**Comment by fabric8cd** [comment]
@xcoulon snapshot fabric8-tenant image is available for testing.  `docker pull docker.io/fabric8/fabric8-tenant:SNAPSHOT-PR-537-5`

File: token/service.go [hunk 3]
```diff
@@ -38,7 +39,7 @@ func (s *tokenService) ResolveUserToken(ctx context.Context, target, token strin
 					Value: token,
 					Type:  "Bearer"}}})
 
-	res, err := client.RetrieveToken(ctx, authclient.RetrieveTokenPath(), target, nil)
+	res, err := client.RetrieveToken(ctx, authclient.RetrieveTokenPath(), target, &forcePull)
 	if err != nil {
 		return "", "", errors.Wrapf(err, "error while doing the request")
 	}
```
[the remark this answers is not in the document]
Comment (aslakknutsen) [reply token/service.go]: I guess a 401 from auth will get caught here https://github.com/xcoulon/fabric8-tenant/blob/a00e36d1244cf92c30f5d61842973429d600cc60/token/service.go#L52

[the remark this answers is not in the document]
Comment (xcoulon) [reply token/service.go]: ok, let me add the `err.Error()` message in the returned error, then.

**Comment by fabric8cd** [comment]
@xcoulon snapshot fabric8-tenant image is available for testing.  `docker pull docker.io/fabric8/fabric8-tenant:SNAPSHOT-PR-537-7`

**Comment by fabric8cd** [comment]
@xcoulon snapshot fabric8-tenant image is available for testing.  `docker pull docker.io/fabric8/fabric8-tenant:SNAPSHOT-PR-537-8`

**Pull request merged**